pkg/providers/instance/instance.go

```diff
@@ -782,20 +827,8 @@ func (p *DefaultProvider) pickSkuSizePriorityAndZone(
 	return nil, "", ""
 }
 
-func (p *DefaultProvider) cleanupAzureResources(ctx context.Context, resourceName string) error {
-	vmErr := deleteVirtualMachineIfExists(ctx, p.azClient.virtualMachinesClient, p.resourceGroup, resourceName)
-	if vmErr != nil {
-		log.FromContext(ctx).Error(vmErr, fmt.Sprintf("virtualMachine.Delete for %s failed", resourceName))
-	}
-	// The order here is intentional, if the VM was created successfully, then we attempt to delete the vm, the
-	// nic, disk and all associated resources will be removed. If the VM was not created successfully and a nic was found,
-	// then we attempt to delete the nic.
-
-	nicErr := deleteNicIfExists(ctx, p.azClient.networkInterfacesClient, p.resourceGroup, resourceName)
-	if nicErr != nil {
-		log.FromContext(ctx).Error(nicErr, fmt.Sprintf("networkinterface.Delete for %s failed", resourceName))
-	}
-	return errors.Join(vmErr, nicErr)
+func (p *DefaultProvider) cleanupAzureResources(ctx context.Context, resourceName string) (err error) {
+	return deleteVirtualMachineIfExists(ctx, p.azClient.virtualMachinesClient, p.resourceGroup, resourceName)
```

Comment:
I was debating keeping in the garbage collection controller in case nic delete fails or NRP isn't succeeding in cleanup. Thoughts?

Reply:
Yes, that makes sense to me. No harm in keeping it for a while, until we have evidence it is not needed.

Reply:
Thinking more on it, the gc controller relies on the existence of the tags 'karpenter.sh', which we no longer are able to put on the network interface with this api.